### PR TITLE
[EPMRPP-38820] Launch logs

### PR DESCRIPTION
**Pull request by ihar-kahadouski**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/commons/querygen/FilterTarget.java [hunk 1]
```diff
@@ -478,7 +481,7 @@ protected void joinTables(SelectQuery<? extends Record> query) {
 					LOG.LOG_MESSAGE,
 					LOG.LAST_MODIFIED,
 					LOG.LOG_LEVEL,
-					LOG.ITEM_ID,
+					LOG.ITEM_ID, LOG.LAUNCH_ID,
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/commons/querygen/FilterTarget.java]: formatting

File: src/main/java/com/epam/ta/reportportal/dao/util/RecordMappers.java [hunk 2]
```diff
@@ -186,17 +186,21 @@
...
+		log.setLogMessage(r.get(LOG.LOG_MESSAGE, String.class));
+		log.setLastModified(r.get(LOG.LAST_MODIFIED, LocalDateTime.class));
+		log.setLogLevel(r.get(JLog.LOG.LOG_LEVEL, Integer.class));
+
+		log.setAttachment(ATTACHMENT_MAPPER.map(r));
+
+		TestItem testItem = new TestItem(r.get(LOG.ITEM_ID));
+		log.setTestItem(testItem);
+
+		Launch launch = new Launch(r.get(LOG.LAUNCH_ID));
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/dao/util/RecordMappers.java]: what if a log doesn't have launch id?

File: src/main/java/com/epam/ta/reportportal/jooq/Keys.java [hunk 3]
```diff
@@ -184,6 +184,7 @@
     public static final ForeignKey<JLdapConfigRecord, JIntegrationRecord> LDAP_CONFIG__LDAP_CONFIG_ID_FKEY = ForeignKeys0.LDAP_CONFIG__LDAP_CONFIG_ID_FKEY;
     public static final ForeignKey<JLdapConfigRecord, JLdapSynchronizationAttributesRecord> LDAP_CONFIG__LDAP_CONFIG_SYNC_ATTRIBUTES_ID_FKEY = ForeignKeys0.LDAP_CONFIG__LDAP_CONFIG_SYNC_ATTRIBUTES_ID_FKEY;
     public static final ForeignKey<JLogRecord, JTestItemRecord> LOG__LOG_ITEM_ID_FKEY = ForeignKeys0.LOG__LOG_ITEM_ID_FKEY;
+	public static final ForeignKey<JLogRecord, JLaunchRecord> LOG__LOG_LAUNCH_ID_FKEY = ForeignKeys0.LOG__LOG_LAUNCH_ID_FKEY;
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/jooq/Keys.java]: formatting

File: src/main/java/com/epam/ta/reportportal/commons/querygen/FilterTarget.java [hunk 4]
```diff
@@ -478,16 +475,15 @@ protected void joinTables(SelectQuery<? extends Record> query) {
...
 	)) {
 		@Override
 		protected Collection<? extends SelectField> selectFields() {
 			return Lists.newArrayList(LOG.ID,
 					LOG.LOG_TIME,
 					LOG.LOG_MESSAGE,
-					LOG.LAST_MODIFIED,
-					LOG.LOG_LEVEL,
-					LOG.ITEM_ID,
+					LOG.LAST_MODIFIED, LOG.LOG_LEVEL, LOG.ITEM_ID, LOG.LAUNCH_ID,
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/commons/querygen/FilterTarget.java]: please, fix your formatting

File: src/main/java/com/epam/ta/reportportal/dao/LogRepositoryCustomImpl.java [hunk 5]
```diff
@@ -123,25 +123,23 @@ public boolean hasLogs(Long itemId) {
 	}
 
 	@Override
-	public List<Long> findIdsByLaunchId(Long launchId) {
+	public List<Long> findItemLogIdsByLaunchId(Long launchId) {
 		return dsl.select()
 				.from(LOG)
 				.leftJoin(TEST_ITEM)
 				.onKey()
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/dao/LogRepositoryCustomImpl.java]: let's make this in common style with 'launch join' below

File: src/main/java/com/epam/ta/reportportal/dao/util/RecordMappers.java (hunk 2 above)
Comment (pbortnik) [reply src/main/java/com/epam/ta/reportportal/dao/util/RecordMappers.java]: ???

Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/dao/util/RecordMappers.java]: fixed

File: src/main/java/com/epam/ta/reportportal/commons/querygen/FilterTarget.java (hunk 1 above)
Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/commons/querygen/FilterTarget.java]: fixed

File: src/main/java/com/epam/ta/reportportal/jooq/Keys.java (hunk 3 above)
Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/jooq/Keys.java]: fixed

File: src/main/java/com/epam/ta/reportportal/commons/querygen/FilterTarget.java (hunk 4 above)
Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/commons/querygen/FilterTarget.java]: fixed

File: src/main/java/com/epam/ta/reportportal/dao/LogRepositoryCustomImpl.java (hunk 5 above)
Comment (ihar-kahadouski) [reply src/main/java/com/epam/ta/reportportal/dao/LogRepositoryCustomImpl.java]: done

**Pull request merged**